pubpub/platform
# feat: make site useable on mobile by making sidenav sheet

This PR makes the sidenav hideable on mobile, and introduces some small changes to it as well.

## Issue(s) Resolved

Almost impossible to use the site on mobile.

## Test Plan

1. Make browser window small enough to be like mobile
2. Try to access each page in `/c/[slug]/` and see if the sidebar works okay

## Concerns

Maybe the margins are a bit too narrow in mobile view, could be slightly more generous.

## Screenshots (if applicable)

### Page is now highlighted in nav

<img width="878" alt="image" src="https://github.com/pubpub/v7/assets/21983833/c08dba2b-d418-4bf3-a8e8-9e01c2ff6e10">

### Mobile view


#### Closed
Introduces a header that contains the switcher and the menu.

Switcher could also be in the sidenav, but it was a bit hard to with the close button in a nice way, and this way you can always see what community you are in.

<img width="568" alt="image" src="https://github.com/pubpub/v7/assets/21983833/0dcae7f8-2b83-4e1e-adbd-79123d334952">


#### Open

<img width="495" alt="image" src="https://github.com/pubpub/v7/assets/21983833/849612c8-8a39-4af6-9436-a3f3e4a8563a">


Even looks fine on the manage page

<img width="495" alt="image" src="https://github.com/pubpub/v7/assets/21983833/039bbacb-300d-417e-b377-ddfc02eae476">


## Optional

### Notes/Context/Gotchas

### Supporting Docs
